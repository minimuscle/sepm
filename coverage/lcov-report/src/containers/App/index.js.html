<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/containers/App/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">src/containers/App</a> index.js
    </h1>
    <div class='clearfix'>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js"><span class="cstat-no" title="statement not covered" >import React, { Component } from 'react';</span>
<span class="cstat-no" title="statement not covered" >import Login from 'containers/Login';</span>
<span class="cstat-no" title="statement not covered" >import './App.css';</span>
<span class="cstat-no" title="statement not covered" >import api from 'api';</span>
<span class="cstat-no" title="statement not covered" >import Main from '../../components/Main/Main';</span>
&nbsp;
class App extends Component {
<span class="fstat-no" title="function not covered" >  constructor</span>(props) <span class="cstat-no" title="statement not covered" >{</span>
<span class="cstat-no" title="statement not covered" >    super(props);</span>
<span class="cstat-no" title="statement not covered" >    this.state = {</span>
      login: false,
      section: 'dashboard',
    };
<span class="cstat-no" title="statement not covered" >    this.enableLogin = this.enableLogin.bind(this);</span>
  }
&nbsp;
  enableLogin = <span class="fstat-no" title="function not covered" >user</span> =&gt; {
<span class="cstat-no" title="statement not covered" >    this.setState({ login: true, user });</span>
  };
&nbsp;
<span class="fstat-no" title="function not covered" >  render</span>() {
<span class="cstat-no" title="statement not covered" >    if (this.state.login) {</span>
<span class="cstat-no" title="statement not covered" >      return &lt;Main user={this.state.user} /&gt;;</span>
    } else <span class="cstat-no" title="statement not covered" >if (!this.state.login) {</span>
<span class="cstat-no" title="statement not covered" >      return &lt;Login onLogin={this.enableLogin} login={api.login} /&gt;;</span>
    }
  }
}
&nbsp;
export default App;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Aug 11 2020 17:15:23 GMT+1000 (Australian Eastern Standard Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
