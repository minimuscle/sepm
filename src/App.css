.title {
    padding: 30px 0px 30px 0px;
}

.addBtn {
    text-align: right;
    float: right;
    margin-top: -50px;
}

.login {
    margin: 100px auto auto auto;
    text-align: center;
}