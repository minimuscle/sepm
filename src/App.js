import React, { Component } from 'react';
import './App.css';
import Tours from './components/Tours/Tours';
import AddTours from './components/Tours/AddTours';
import EditTours from './components/Tours/EditTours';
import Types from './components/TourTypes/Types';
import AddTypes from './components/TourTypes/AddTypes';
import EditTypes from './components/TourTypes/EditTypes';
import Locations from './components/Locations/Locations';
import AddLocations from './components/Locations/AddLocations';

//Boostrap Imports - Design work
import { Navbar, Nav, Container } from 'react-bootstrap';


class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      view: 'locations',
      name: ''
    }
  }

  updateView(view) {
    this.setState({
      view: view
    })
  }

  setName(name) {
    this.setState({
      name: name
    })
  }


  renderComponents() {
    switch (this.state.view) {
      case 'tours':
        return <Tours view={this.updateView.bind(this)} edit={this.setName.bind(this)}/>

      case 'edit-tour':
        return <EditTours view={this.updateView.bind(this)} title={this.state.name}/>

      case 'types':
        //The bind sets the 'view' to the variable that is set in <Types/>
        return <Types view={this.updateView.bind(this)} edit={this.setName.bind(this)}/>

      case 'add-tour':
        return <AddTours />

      case 'add-location':
        return <AddLocations />

      case 'locations':
        return <Locations view={this.updateView.bind(this)}/>

      case 'add-type':
        return <AddTypes />

      case 'edit-type':
        return <EditTypes view={this.updateView.bind(this)} title={this.state.name}/>
      //Default means that if there is an error or not a 'case' then it defaults to the tours page
      default:
        return <Locations view={this.updateView.bind(this)} edit={this.setName.bind(this)}/>
    }
  }

  render() {
    return (
      <Container fluid>
        <Navbar bg="dark" variant="dark">
          {/** TODO: Change other tours to be their actual pages */}
          <Navbar.Brand onClick={() => this.updateView('tours')}>Tour Guide Manager</Navbar.Brand>
          <Navbar.Collapse>
            <Nav className='mr-auto'>
              <Nav.Link onClick={() => this.updateView('locations')}>Locations</Nav.Link>
              <Nav.Link onClick={() => this.updateView('tours')}>Tours</Nav.Link>
              <Nav.Link onClick={() => this.updateView('types')}>Tour Types</Nav.Link>
            </Nav>
          </Navbar.Collapse>
          <Navbar.Text>
            Signed in as <a href="#login">John Doe</a>
          </Navbar.Text>
        </Navbar>
        {this.renderComponents()}
      </Container>
    )
  }
}

export default App;
