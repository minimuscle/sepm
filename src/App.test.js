import App from './App'
import { shallow, mount } from 'enzyme';
import React from 'react';

//Sidebar exists
test("App to exist", () => {
    expect(App).toMatchSnapshot();
})

