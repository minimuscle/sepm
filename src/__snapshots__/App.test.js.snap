// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App to exist 1`] = `[Function]`;
